py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,135 @@
+
+SAMPLE OUTPUT
+{'color': '#00FF00', 'full_text': u'Bouygues Telecom 4G (19%)'}
+
+off
+{'color': '#FF0000', 'full_text': u'Bouygues Telecom 4G (12%)'}
+"""
+
+from pydbus import SystemBus
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    cache_timeout = 5
+    consider_3G_degraded = False
+    format_down = 'WWAN: {operator} {netgen} ({signal})'
+    format_error = 'WWAN: {error}'
+    format_up = 'WWAN: {operator} {netgen} ({signal})'
+    """
+    modem: 'auto' value will use first find modem or
+    use 'busctl introspect org.freedesktop.ModemManager1 \
+            /org/freedesktop/ModemManager1/Modem/0'
+    and read '.EquipmentIdentifier')
+    """
+    modem = 'auto'
+
+    def wwan_status_nm(self):
+        response = {}
+        response['cached_until'] = self.py3.time_in(self.cache_timeout)
+
+        bus = SystemBus()
+        for id in range(0, 20):
+            """
+            find the modem
+            """
+            device = 'Modem/' + str(id)
+
+            try:
+                proxy = bus.get('.ModemManager1', device)
+
+                eqid = str(proxy.EquipmentIdentifier)
+
+                if (self.modem != 'auto'
+                        and eqid == self.modem) or (self.modem == 'auto'):
+                    state = proxy.State
+                    signal = str(proxy.SignalQuality[0]) + '%'
+                    modes = proxy.CurrentModes[0]
+                    operator = proxy.OperatorName
+                    netgen = self._get_capabilities(modes)
+
+                    if state == 11:
+                        response['full_text'] = self.py3.safe_format(
+                            self.format_up,
+                            dict(
+                                netgen=netgen[1],
+                                signal=signal,
+                                operator=operator))
+                        """
+                        green color if 4G, else yellow
+                        """
+                        if netgen[0] == 4:
+                            response['color'] = self.py3.COLOR_GOOD
+                        elif netgen[0] == 3 and self.consider_3G_degraded is False:
+                            response['color'] = self.py3.COLOR_GOOD
+                        else:
+                            response['color'] = self.py3.COLOR_DEGRADED
+
+                    else:
+                        response['full_text'] = self.py3.safe_format(
+                            self.format_down,
+                            dict(
+                                netgen=netgen[1],
+                                signal=signal,
+                                operator=operator))
+                        response['color'] = self.py3.COLOR_BAD
+                else:
+                    response['full_text'] = self.py3.safe_format(
+                        self.format_error, dict(error='wrong or empty modem'))
+                    response['color'] = self.py3.COLOR_BAD
```

Comment:
We likely want to use `self.py3.error()` here. See `bluetooth` or `external_script`.